ansible/roles/nginx/templates/nginx.conf.j2

```diff
@@ -89,20 +89,27 @@ http {
 {# Turn off sending information about the server to the client #}
     server_tokens off;
 
+    # Redirect all http to https.
+    server {
+        listen 80 default_server;
+        server_name _;
+        return 301 https://$host$request_uri;
```

Comment:
In some down streams, they might want to use http for some performance reason.
How about making this redirection configurable?

Reply:
I could do that but note that port 80 wasn’t open at all - so one would have had to provide their own nginx anyway?

Reply:
ok, I just thought they might take advantage of this change.
(by just using upstream default conf)

Reply:
i can remove this if it's controversial -- my thinking was that we have not supported port 80 but the port was exposed... so this change redirect http to https.

i think the reason not to support http in the project is that it wouldn't/shouldn't be used in production, and we've had support for self signed cert for local development, and without a test it would rot.

incidentally, i could add a test for the http redirect.